### PR TITLE
[gui] Extend settings dialog functions to show seperator with group name (small version)

### PR DESCRIPTION
**Pull request by AlwinEsch**
This is the reduced version of the request #6502, which have the for AudioDSP needed parts.

bildschirmfoto8

### CODE REVIEW

**Comment by ronie** [comment]
great. clean & simple, just the way i like it :-)

**Comment by phil65** [comment]
yes, looks great. thx Alwin :)

File: system/settings/settings.xml [hunk 1]
```diff
@@ -91,7 +91,7 @@
           <control type="list" format="string" />
         </setting>
       </group>
-      <group id="2">
+      <group id="2" label="371" help="-1">
```
Comment (da-anda) [inline system/settings/settings.xml]: do we really need the help="-1" stuff in here?

**Comment by da-anda** [comment]
thanks for not giving up on this :) Fine with me.

Side note: I still don't like that the code decides if certain things are shown in a skin or not, especially stuff like "hideSeparator" or "labelBeforeSeparator", because this decision is IMO not in the domain of the controlling logic, but only the skin. Some skins don't even have/show a separator, so any design related decision done by a dev anywhere is totally out of place. But that's not just an issue with this PR but IMO a more general one in our GUI implemenation.

File: system/settings/settings.xml (hunk 1 above)
Comment (AlwinEsch) [inline system/settings/settings.xml]: I'm not sure has seen that all other stuffs has used it and thought to add it also there. But from my side no help needed for groups, is only to reduce settings label size, I remove it.

**Comment by AlwinEsch** [comment]
Hi @da-anda, I do not give up so fast :-)

File: system/settings/settings.xml (hunk 1 above)
Comment (da-anda) [inline system/settings/settings.xml]: don't get me wrong - the ability to assign a help string in general is nice (and also needed for your ADSP stuff IIRC), but defining this attribute here when it's not used in this case didn't make sense.

File: xbmc/settings/lib/SettingSection.h [hunk 2]
```diff
@@ -50,6 +50,34 @@ class CSettingGroup : public ISetting
...
+  bool LabelBeforeSeparator() { return m_labelBeforeSeparator; }
+  /*!
+   \brief Sets the position of localizeable label ID of the setting group.
+   behind or before a separator.
+
+   \param before The wanted position
+   */
+  void SetLabelBeforeSeparator(bool before) { m_labelBeforeSeparator = before; }
+  /*!
+   \brief Gets the position of localizeable label ID of the setting group.
```
Comment (xhaggi) [inline xbmc/settings/lib/SettingSection.h]: c/p wrong documentation

File: xbmc/settings/lib/SettingSection.h [hunk 3]
```diff
@@ -50,6 +50,34 @@ class CSettingGroup : public ISetting
...
+   behind or before a separator.
+
+   \return The wanted position
+   */
+  bool HideSeparator() { return m_HideSeparator; }
+  /*!
+   \brief Sets the position of localizeable label ID of the setting group.
```
Comment (xhaggi) [inline xbmc/settings/lib/SettingSection.h]: same here

Comment (AlwinEsch) [inline xbmc/settings/lib/SettingSection.h]: Sorry, is fixed

**Comment by Montellese** [comment]
Sorry for the late reply.

There's one thing I would like to change and that is the GUI-related properties you added to `CSettingGroup` in `xbmc/settings/lib/SettingSection.h`. Right now the whole code in `xbmc/settings/lib/` is completely decoupled from anything GUI related except for the `ISettingControl` interface which gives the application the possibility to integrate GUI information into the settings definition. I'd prefer it if we do the same for `CSettingGroup`. So either duplicate the logic from `CSetting` to handle an `ISettingControl` member or move that logic to `ISetting` as well. Then you need to add a new implementation (e.g. `CSettingControlTitle`) derived from `ISettingControl` to `xbmc/settings/SettingControl.h` and integrate that into the rest.

That way the resulting setting definition in XML would look like this
```xml
<section ...>
  <category ...>
    <group id="1" help="2">
      <control type="title">
        <hideseparator>false</hideseparator>
        <separatorposition>bottom</separatorposition>
      </control>
      <setting ...>
        ...
      </setting>
      ...
    </group>
    ...
  </category>
  ...
</section>
...
```
or something like that.

**Comment by AlwinEsch** [comment]
Sorry for my late reply was a bit sick the days before.
About the label position, I think now we can leave it in this version on a fixed position and I remove the before/behind, for dsp is only the hide separator relevant.
```xml
 <group id="1" label="2" hideseparator="false">
```
What do you think?

**Comment by Montellese** [comment]
No that would still put GUI related code into the settings library which I don't want. The approach I proposed would still be the same even if you remove the label position configuration (which IMO doesn't really make much sense anyway).

**Comment by AlwinEsch** [comment]
Thanks a lot @Montellese,
is updated and use your idea. Has added again a temporary test commit to see how in settings.xml.

**Comment by Montellese** [comment]
You misunderstood me. The logic to deserialize the GUI related properties and the setters and getters for these properties must not be part of `CSettingGroup` (or any other other class of the settings library for that matter). Please take a look how `ISettingControl` implementations (all defined in `xbmc/settings/SettingControl.h`) are parsed and handled in `CSetting` et al. The new `title` control should be just another of these `ISettingControl` implementations and handled the same as all others.

Another note: `separatorBehindLabel` sounds wrong and should probably be `separatorBelowLabel` but I thought you wanted to drop that property completely?

**Comment by AlwinEsch** [comment]
Sorry @Montellese, I misunderstood you. I remove it complete from CSettingGroup an bring in this:
```C++
class CSettingControlLabel : public ISettingControl
{
public:
  CSettingControlLabel()
    : m_hidden(false),
      m_heading(-1) { }
  virtual ~CSettingControlEdit() { }

  // implementation of ISettingControl
  virtual std::string GetType() const { return "label"; }
  virtual bool Deserialize(const TiXmlNode *node, bool update = false);
  virtual bool SetFormat(const std::string &format);

  bool IsHidden() const { return m_hidden; }
  void SetHidden(bool hidden) { m_hidden = hidden; }
  int GetHeading() const { return m_heading; }
  void SetHeading(int heading) { m_heading = heading; }

protected:
  bool m_hidden;
  int m_heading;
};
```

**Comment by AlwinEsch** [comment]
So, updated again to the right style (I hope :-) ).

**Comment by Montellese** [comment]
A few minor things but it looks much better now. Could you split your commits into
* changes to the settings library (i.e. anything under `xbmc/settings/lib/`
* changes to the settings integration code (i.e. anything under `xbmc/settings/`
* changes to confluence

**Comment by AlwinEsch** [comment]
Is updated, parts renamed and separated to several commits, thanks @Montellese

**Comment by Montellese** [comment]
Cool thanks. Only three minors and then this should be good to go.

**Comment by Montellese** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
Has pushed again and is now up to date, not sure what the fault before was, but the last changes was not seen there, but now is in.

jenkins build this please again

**Comment by Montellese** [comment]
Thanks, looks good for me now. @ronie @phil65 any objections?

**Comment by ronie** [comment]
it's fine with me.

**Comment by phil65** [comment]
fine for me, too.

**Comment by AlwinEsch** [comment]
Thanks a lot to all, Juhu :-D.

**Pull request merged**